src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyTypeFlag.cs

```diff
@@ -40,7 +41,9 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return StringComparer.Ordinal.GetHashCode(_value);
+            var combiner = new HashCodeCombiner();
```

Comment:
What is the benefit of changing for HashCodeCombiner in this context?
Isn't just `StringComparer.Ordinal.GetHashCode` enough for simple string?

Reply:
Have a look at the remarks for the docs on `string.GetHashCode()`: https://docs.microsoft.com/en-us/dotnet/api/system.string.gethashcode?view=netcore-3.1

Particularly this bit:

> The hash code itself is not guaranteed to be stable. Hash codes for identical strings can differ across .NET implementations, across .NET versions, and across .NET platforms (such as 32-bit and 64-bit) for a single version of .NET. In some cases, they can even differ by application domain. This implies that two subsequent runs of the same program may return different hash codes.
> 
> As a result, hash codes should never be used outside of the application domain in which they were created, they should never be used as key fields in a collection, and they should never be persisted.

Classes used in no-op restore calculations have a hash persisted to disk. I'm not sure if it's the `GetHashCode` that is persisted, or some other hash, but we do persist a hash to disk to compare against to determine no-op restore or if a "full" restore is required. Hence it's critical to have a stable hash.

Reply:
It's not used for persisting yet, but the stable implementation allows us to consider it in a future work item, like say https://github.com/NuGet/Home/issues/10068